crates/router/src/connector/iatapay/transformers.rs

```
@@ -0,0 +1,369 @@
+                token: item.response.access_token,
+                expires: item.response.expires_in,
+            }),
+            ..item.data
+        })
+    }
+}
+
+#[derive(Default, Debug, Serialize, Eq, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct RedirectUrls {
+    success_url: String,
+    failure_url: String,
+}
+
+//TODO: Fill the struct with respective fields
+#[derive(Default, Debug, Serialize, Eq, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct IatapayPaymentsRequest {
+    merchant_id: String,
+    amount: i64,
+    currency: String,
+    country: String,
+    locale: String,
+    redirect_urls: RedirectUrls,
+    notification_url: String,
+}
+
+#[derive(Default, Debug, Serialize, Eq, PartialEq)]
+pub struct IatapayCard {
+    name: Secret<String>,
+    number: Secret<String, common_utils::pii::CardNumber>,
+    expiry_month: Secret<String>,
+    expiry_year: Secret<String>,
+    cvc: Secret<String>,
+    complete: bool,
+}
+
+impl TryFrom<&types::PaymentsAuthorizeRouterData> for IatapayPaymentsRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &types::PaymentsAuthorizeRouterData) -> Result<Self, Self::Error> {
+        let country = item.get_billing_country()?.to_string();
+        let payload = Self {
+            merchant_id: IatapayAuthType::try_from(&item.connector_auth_type)?.merchant_id,
+            amount: item.request.amount,
+            currency: item.request.currency.to_string(),
+            country: country.clone(),
+            locale: format!("en-{}", country),
+            redirect_urls: get_redirect_url(item),
+            notification_url: item.request.get_webhook_url()?,
+        };
+        Ok(payload)
+    }
+}
+
+fn get_redirect_url(item: &types::PaymentsAuthorizeRouterData) -> RedirectUrls {
+    let return_url = item
+        .request
+        .get_return_url()
+        .unwrap_or("https://hyperswitch.io".to_string());
```

let it throw the error why do you hardcode the return_url

---

done